central/reports/scheduler/schedule.go

```diff
@@ -364,24 +385,42 @@ func formatMessage(rc *storage.ReportConfiguration, emailTemplate string, date t
 }
 
 func (s *scheduler) getReportData(ctx context.Context, rQuery *common.ReportQuery) ([]common.Result, error) {
-	r := make([]common.Result, 0, len(rQuery.ScopeQueries))
-	for _, sq := range rQuery.ScopeQueries {
-		resultData, err := s.runPaginatedQuery(ctx, sq, rQuery.CveFieldsQuery)
-		if err != nil {
-			return nil, err
+	if !features.ObjectCollections.Enabled() {
+		r := make([]common.Result, 0, len(rQuery.ScopeQueries))
+		for _, sq := range rQuery.ScopeQueries {
+			resultData, err := s.runPaginatedQuery(ctx, sq, rQuery.CveFieldsQuery, rQuery.DeploymentsQuery)
+			if err != nil {
+				return nil, err
+			}
+			r = append(r, resultData)
 		}
-		r = append(r, resultData)
+		return r, nil
 	}
-	return r, nil
+	result, err := s.runPaginatedQuery(ctx, "", rQuery.CveFieldsQuery, rQuery.DeploymentsQuery)
+	if err != nil {
+		return nil, err
+	}
+	result = groupByClusterAndNamespace(result)
+	return []common.Result{result}, nil
 }
 
-func (s *scheduler) runPaginatedQuery(ctx context.Context, scopeQuery, cveQuery string) (common.Result, error) {
+func (s *scheduler) runPaginatedQuery(ctx context.Context, scopeQuery, cveQuery string, deploymentsQuery *v1.Query) (common.Result, error) {
 	offset := 0
 	var resultData common.Result
 	for {
 		var gqlQuery string
+		gqlPaginationOffset := offset
 		if env.PostgresDatastoreEnabled.BooleanSetting() {
 			gqlQuery = reportQueryPostgres
+			if features.ObjectCollections.Enabled() {
+				deploymentIds, err := s.getDeploymentIDs(ctx, deploymentsQuery, int32(offset))
+				if err != nil || len(deploymentIds) == 0 {
+					return common.Result{}, err
+				}
+				scopeQuery = fmt.Sprintf("%s:%s", search.DeploymentID.String(), strings.Join(deploymentIds, ","))
```

Comment:
Why do you need to fetch deployment IDs? You can pass deployment query to deployment resolver.

Reply:
deploymentQuery is a `*v1.Query` but graphQL needs a string query. The problem with building an equivalent string version of  deploymentQuery occurs with nested collections. Nested collections will  form queries like  
`[(Cluster: c1 AND Namespace: n1 AND Deployment: d1) OR (Cluster: c2 AND Namespace: n2 AND Deployment: d2)]` 

However, we cannot make a similar string query of the form 
`scopeQuery: (Cluster:c1+Namespace:n1+Deployment:d1) OR (Cluster:c2+Namespace:n2+Deployment:d2)` because the string query language doesn't support disjunction of conjunctions. It only supports conjunction of disjunctions where each fieldLabel can only occur once. This is why I need to fetch deploymentIDs first and then create a scopeQuery from those to pass to graphQL.

Reply:
Add the explanation as code comment